pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BaseReduceService.java

```diff
@@ -318,7 +318,7 @@ protected void setStats(String rawTableName, BrokerResponseNative brokerResponse
         brokerMetrics.addTimedTableValue(rawTableName, BrokerTimer.REALTIME_TOTAL_CPU_TIME_NS, _realtimeTotalCpuTimeNs,
             TimeUnit.NANOSECONDS);
 
-        if (_numConsumingSegmentsProcessed > 0 && _minConsumingFreshnessTimeMs > 0) {
+        if (_minConsumingFreshnessTimeMs != Long.MAX_VALUE) {
```

Comment:
Why removing the first part of the check?

Reply:
_minConsumingFreshnessTimeMs may have value even when _numConsumingSegmentsProcessed equals 0, in some error scenarios. Previously _numConsumingSegmentsProcessed > 0 was used in a way to check if consuming segments are present and emit metrics only when the condition is reached. Now we want to emit metrics even if no consuming segments are present, so the first part becomes redundant and can be removed

Reply:
@PrachiKhobragade  We can probably put it as the following to exactly show what we would like to achieve?
```
if (_numConsumingSegmentsProcessed > 0 || (_numConsumingSegmentsProcessed == 0 && _minConsumingFreshnessTimeMs != Long.MAX_VALUE)) {
 ...
}
```

```

Reply:
I see, the current code is okay. Is it possible that server returns a non-positive `minConsumingFreshnessTimeMs` in the current code? If so, this change might be backward incompatible; if not, then the change is good.

Reply:
The change in behavior of `minConsumingFreshnessTimeMs` is update it to maxEndTIme if `numConsumingSegments == 0` maxEndTIme is end of the segment timeInterval, so should never be negative. 

we update the `minConsumingFreshnessTimeMs` only when `maxEndTIme !=  LONG.MIN` so `minConsumingFreshnessTimeMs` shouldn't get a -ve value